### PR TITLE
[FIX] hr_holidays: manage accrual plan with postpone max days

### PR DESCRIPTION
**Pull request by thle-odoo**
Steps to reproduce:
-------------------
- create an accrual plan with `postpone_max_days` not equal to 0;
- create an allocation with this accrual plan with a date_from in the last year;
- to be in the following year;
- run the cron to update the accrual plan allocations.

Issue:
------
We do not receive the expected total allocation.

Cause:
------
We force the lastcall date to the first day of this year. This is correct under the following assumption:
"The allocations between last year's lastcall and
the first day of this year have indeed been made".

In this case, this is not true.
In fact, as the allocation does not have a `nextcall`, it will be calculated from its lastcall (for example 25 December 2023), which will give us a date greater than the date on which we want to force the nextcall (for example 25 January 2024 > 1 January 2024).

As a result, we don't enter the loop that calculates the allocations to be assigned. But as we are forcing the lastcall on the first day of the year, the allocations calculated between 25 December 2023 and 1 January 2024 in this case will never be allocated.

Solution:
---------
We don't need to force lastcall on the first day of the year as lastcall is the old nextcall.

When the loop that calculates the allocations to be made is executed with a nextcall equal to the first day of the year, the lastcall will be updated with this value (and at this point we can be sure that we can do it because the days corresponding to a period in the past year will be allocated).

Note:
-----
Correction of the fix: 60ee2dc2601c1fe7440d69d01844cc7eca4d39df

opw-3704690

### CODE REVIEW

File: addons/hr_holidays/tests/test_accrual_allocations.py [hunk 1]
```diff
@@ -909,6 +909,44 @@ def test_unused_accrual_postponed_limit(self):
             allocation._update_accrual()
         self.assertEqual(allocation.number_of_days, 22, 'The maximum number of days should be reached and kept.')
 
+    def test_unused_accrual_postponed_limit_without_nextcall(self):
+        hr_leave_allocation = self.env['hr.leave.allocation']
```
Comment (Bertrand2) [inline addons/hr_holidays/tests/test_accrual_allocations.py:913]:
```suggestion
```

File: addons/hr_holidays/tests/test_accrual_allocations.py [hunk 2]
```diff
@@ -909,6 +909,44 @@ def test_unused_accrual_postponed_limit(self):
...
+            })
+            allocation_postponed.action_confirm()
+            allocation_postponed.action_validate()
+
+        accrual_cron = self.env['ir.cron'].sudo().env.ref('hr_holidays.hr_leave_allocation_cron_accrual')
+        accrual_cron.lastcall = datetime.date(2023, 12, 25)
+
+        with freeze_time('2024-2-10'):
+            # allocation_postponed.nextcall = False
+            hr_leave_allocation._update_accrual()
```
Comment (Bertrand2) [inline addons/hr_holidays/tests/test_accrual_allocations.py:945]:
```suggestion
            self.env['hr.leave.allocation']._update_accrual()
```

**Comment by thle-odoo** [comment]
Note:
We have to believe what's happening in the loop (`while allocation.nextcall <= date_to` in `_process_accrual_plans`) because that's when the leave is allocated. Do not force lastcall and nextcall after the loop. Before if we want, but not after.

**Review by Bertrand2 — APPROVED** [review]
Nitpicking, otherwise LGTM, thanks!

**Pull request closed without merging**